lambdas/src/handlers/access-token-handler.ts

```diff
@@ -29,30 +30,27 @@ export class AccessTokenLambda implements LambdaInterface {
         private readonly sessionService: SessionService,
         private readonly requestValidator: AccessTokenRequestValidator,
     ) {}
-
-    @logger.injectLambdaContext({ clearState: true })
     @metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
     @_tracer.captureLambdaHandler({ captureResponse: false })
-    public async handler(event: APIGatewayProxyEvent, _context: unknown): Promise<APIGatewayProxyResult> {
+    public async handler(event: APIGatewayProxyEvent, _context: Context): Promise<APIGatewayProxyResult> {
         try {
-            await initPromise;
             logger.info("Access Token Lambda triggered");
-
-            const requestPayload = this.requestValidator.validatePayload(event.body);
-            const sessionItem = await this.sessionService.getSessionByAuthorizationCode(requestPayload.code);
-            logger.info("Session found");
+            const eventBody = event.body;
+            const sessionItem = eventBody as unknown as SessionItem;
+            const requestPayload = eventBody as unknown as RequestPayload;
```

Comment:
Is it worth doing something like `const item = eventBody as unknown as NewType` where NewType is a combo of SessionItem and RequestPayload

Reply:
I'm not 100% that it'll work but casting the event as two different types feels a bit hacky

Reply:
Like you've used below `const event_body = request.event.body as SessionItem & RequestPayload;`

Reply:
I did what you proposed initially, but decided against it. Because it's then no longer obvious which fields come from where when looking at the lambda handler code. I tested the above and it worked fine.